plugins/templates/create.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/templates',
+    config: {
+        description: 'Create a new template',
+        notes: 'Create a specific template',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['build']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const templateFactory = request.server.app.templateFactory;
+            const token = request.auth.credentials;
+            const pipelineId = token.pipelineId;
+            const name = request.payload.name;
+            const labels = request.payload.labels || [];
+
+            return Promise.all([
+                pipelineFactory.get(pipelineId),
+                templateFactory.get({ name })
```

Comment:
Thanks to @FenrirUnbound 
This is because https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L75

It expects all the keys to be there.